code/src/OSFramework/Grid/AbstractDataSource.ts

```diff
@@ -0,0 +1,222 @@
+        dataArray.forEach((item) => {
+            returnDataArray.push(OSFramework.Helper.Flatten(item));
+        });
+
+        return returnDataArray;
+    }
+
+    /**
+     * Responsible for parsing string fields to its correct data type
+     *
+     * @param data A string of the JSON data object
+     * @example OS Date fields is send as strings, and we should parse it to Date, the is done for Datetime fields.
+     */
+    function ToJSONFormat(
+        data: string,
+        convertions: Map<string, Set<string>>
+        // eslint-disable-next-line @typescript-eslint/naming-convention, @typescript-eslint/no-explicit-any
+    ): any {
+        //regex expressions for date and datetime should be described here
+        const regex = {
+            datetime: /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2}(?:\.\d*)?)Z?$/, //yyyy-MM-ddThh:mm:ssZ
+            date: /^(\d{4})-(\d{2})-(\d{2})$/ //yyyy-MM-dd
+        };
+
+        // Parse dates saved as JSON-strings
+        return JSON.parse(data, (key, value) => {
+            if (typeof value === 'string') {
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                let m: any;
+
+                const match = (value: string, exp: RegExp) => {
+                    m = value.match(exp);
+                    return m;
+                };
+
+                const saveConvertion = (type: string, key: string) => {
+                    const done = convertions.get(type) || new Set<string>();
+                    done.add(key);
+                    convertions.set(type, done);
+                };
+
+                if (match(value, regex.datetime)) {
+                    saveConvertion('datetime', key);
+                    return new Date(
+                        Date.UTC(+m[1], +m[2] - 1, +m[3], +m[4], +m[5], +m[6])
+                    );
+                } else if (match(value, regex.date)) {
+                    //Considering that OS Date field do not consider GMT
+                    //DataGrid also won't consider it for Date Columns
+                    //PS: Datetime will consider GMT just like OS consider
+                    saveConvertion('date', key);
+                    return new Date(+m[1], +m[2] - 1, +m[3]);
+                } else if (value === '') {
+                    return undefined;
+                }
+            }
+            return value;
+        });
+    }
+
+    /**
+     * Used to send data to Outsystems, converting data to OS format.
+     * @param grid Grid to format data
+     * @param data Data to format
+     */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    function ToOSFormat(
+        convertions: Map<string, Set<string>>,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        data: Array<any>
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ): void {
+        //TODO: [RGRIDT-638] Regression 2021-02-12: Is this method the best solution
```

Comment:
Maybe... just maybe... let's no leave this comment here!?! :D

Reply:
This //TODO was in the code already... could you fix it for me? =D